### PR TITLE
executor: fix APPROX_PERCENTILE unstable panic when parameter be P100

### LINKED ISSUE #26807: select return error:ERROR 1105 (HY000): runtime error:“ integer divide by zero occasionally” or ”index out of range“
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
CREATE TABLE `tbl_35` (
  `col_244` smallint(6) NOT NULL DEFAULT '-12567',
  `col_245` bigint(20) unsigned DEFAULT NULL,
  `col_246` tinyint(4) NOT NULL,
  `col_247` tinyint(1) NOT NULL,
  `col_248` mediumint(8) unsigned DEFAULT NULL,
  `col_249` bigint(20) NOT NULL,
  `col_250` tinyint(3) unsigned NOT NULL DEFAULT '52',
  `col_251` int(10) unsigned NOT NULL DEFAULT '3672190216',
  PRIMARY KEY (`col_251`,`col_249`) /*T![clustered_index] CLUSTERED */,
  KEY `idx_82` (`col_250`,`col_248`,`col_251`,`col_244`),
  UNIQUE KEY `idx_83` (`col_248`,`col_250`,`col_247`,`col_246`,`col_244`,`col_249`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci
PARTITION BY HASH( `col_249` )
PARTITIONS 6；

insert into tbl_35 values(-12567,2039475165784309258,  12,1,  295431, 5108568534753042951, 52,3672190216);
insert into tbl_35 values( 10921,7383050426504683878, -24,1,  575644,-8481892470407815658, 69,1178171067);
insert into tbl_35 values(-12567,  31055430965522696,  89,1,15938351,-5305129692297068462, 52,3672190216);
insert into tbl_35 values( 16990,6364109974099788726,  45,1, 6085727, 4046315604437132474, 52,3672190216);
insert into tbl_35 values(-30627,   6587933123738203, -59,1, 4818297,  219835403999226661,195, 571380197);
insert into tbl_35 values(-12567, 420855916349948304,   4,0, 5333525,-7854584778698567131, 52,3672190216);
insert into tbl_35 values(  1261,8560887231683292749,-112,0, 1845818,-7456880248953153031, 52,3672190216);
insert into tbl_35 values(-12567,4101176173470815398, -20,0, 7736327, -879859309190038219, 52,3672190216);
insert into tbl_35 values(-12567,4710177654909059880,  11,1, 2925302, 1284439958228355577, 52,3672190216);
insert into tbl_35 values(-23584,4718069391352853238, -89,1,  451974, 1507620968110739479, 52,3672190216);
insert into tbl_35 values(-12567,1990284829886584698, -89,0,   99459, 1875855323515394959, 52,3672190216);
insert into tbl_35 values( 24508,6778324720546396927, -92,1,    NULL,-2450297623516827509, 52,3672190216);
insert into tbl_35 values( 23555,3290440220320110711,  68,1,14307176, -165079195802779661, 52,3672190216);
insert into tbl_35 values(-12567,6976609217105510486, -32,1,12696332, 2135258411092839407, 52,3672190216);
insert into tbl_35 values(-26628,2510884000352962361, -30,0, 7800286, 3287043294638418065, 52,3672190216);
insert into tbl_35 values(-12567,4408923544517225002,  20,1,  194353, 3445749248693394665, 52,3672190216);
insert into tbl_35 values(-12567,5083946274055385841, -24,0,13950880, 7463945667013917749, 52,3672190216);
insert into tbl_35 values(-12567,6230179212097425583,  88,0, 8035271, 7555444438016890289, 52,3672190216);
insert into tbl_35 values(-19438, 937153351609097690,  22,0, 7800522, -524450175229525540,126,2658864355);
insert into tbl_35 values(-12567,7854999488979625038,-109,1,15493020,-3866114964040071826, 52,3672190216);
insert into tbl_35 values(-12567,5880030640634864989,  28,0, 2719798, 1208590981860196858, 52,3672190216);
insert into tbl_35 values(-15088,5015524994767041146,  60,1, 1943533, 2233557247694087902, 52,3672190216);
insert into tbl_35 values(-12567,7670541236326225813, -83,1, 8537665, 2252578468814218504, 52,3672190216);
insert into tbl_35 values(-12567,3916467909459577005,  37,1,14079968, 4391363789796645970, 52,3672190216);
insert into tbl_35 values(-12567, 339507939220463717,  20,1, 4486826, 5394720489477460606, 52,3672190216);
insert into tbl_35 values(-12567,8041435190777307590,   1,0,11817423,-8797651559769414378, 52,3672190216);
insert into tbl_35 values(  9001,3021671509253755664,  17,0,14495370,-6457616006985138726, 52,3672190216);
insert into tbl_35 values(-12567,1763716571886143420, 123,0, 3832916,-5230555791873807348, 52,3672190216);

select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 ) or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol;

```

### 2. What did you expect to see? (Required)

```
mysql> select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 )
or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol;
+--------+
| aggCol |
+--------+
|    195 |
+--------+
1 row in set, 1 warning (0.01 sec)
```

### 3. What did you see instead (Required)

```
mysql> select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 )
or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol;
+--------+
| aggCol |
+--------+
|    195 |
+--------+
1 row in set, 1 warning (0.01 sec)

mysql> select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 )
or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol;
ERROR 1105 (HY000): runtime error: index out of range [28] with length 28
mysql> select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 )
or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol;
ERROR 1105 (HY000): runtime error: integer divide by zero
mysql> select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 )
or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol;
ERROR 1105 (HY000): runtime error: integer divide by zero
mysql> select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 )
or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol;
+--------+
| aggCol |
+--------+
|    195 |
+--------+
1 row in set, 1 warning (0.01 sec)
```

tidb log:

1、index out of range [28] with length 28:

```
[2021/08/03 01:33:09.963 +00:00] [INFO] [conn.go:995] ["command dispatched failed"] [conn=9057] [connInfo="id:9057, addr:192.168.122.1:5823 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 ) or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol"] [txn_mode=PESSIMISTIC] [err="runtime error: index out of range [28] with length 28\ngithub.com/pingcap/tidb/executor.(*recordSet).Next.func1\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/adapter.go:141\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:965\nruntime.goPanicIndex\n\t/usr/local/go/src/runtime/panic.go:88\ngithub.com/pingcap/tidb/executor/aggfuncs.partialResult4PercentileInt.Swap\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggfuncs/func_percentile.go:76\ngithub.com/pingcap/tidb/util/selection.partitionIntro\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:137\ngithub.com/pingcap/tidb/util/selection.medianOfMedians\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:74\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:40\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.Select\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:27\ngithub.com/pingcap/tidb/executor/aggfuncs.percentile\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggfuncs/func_percentile.go:43\ngithub.com/pingcap/tidb/executor/aggfuncs.(*percentileOriginal4Int).AppendFinalResult2Chunk\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggfuncs/func_percentile.go:172\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).appendResult2Chunk\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1376\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).consumeCurGroupRowsAndFetchChild\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1359\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).consumeOneGroup\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1296\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1266\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/executor.go:285\ngithub.com/pingcap/tidb/executor.(*SortExec).fetchRowChunks\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/sort.go:228\ngithub.com/pingcap/tidb/executor.(*SortExec).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/sort.go:112\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/executor.go:285\ngithub.com/pingcap/tidb/executor.(*recordSet).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/adapter.go:145\ngithub.com/pingcap/tidb/server.(*tidbResultSet).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/driver_tidb.go:305\ngithub.com/pingcap/tidb/server.(*clientConn).writeChunks\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1993\ngithub.com/pingcap/tidb/server.(*clientConn).writeResultset\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1941\ngithub.com/pingcap/tidb/server.(*clientConn).handleStmt\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1835\ngithub.com/pingcap/tidb/server.(*clientConn).handleQuery\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1681\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1215\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:978"]
```

2、runtime error: integer divide by zero:

```
[2021/08/03 01:30:08.685 +00:00] [INFO] [conn.go:995] ["command dispatched failed"] [conn=9057] [connInfo="id:9057, addr:192.168.122.1:5823 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select /*+ agg_to_cop() stream_agg() */ approx_percentile( col_250 , 100 ) aggCol from (select   * from tbl_35 t where not( IsNull( t.col_249 ) ) or t.col_246 not in ( -93 ) or not( t.col_245 between 1861738122396168665 and 6503630781418033805 ) or t.col_245 <= 3420550844659577896 order by col_251,col_249  ) ordered_tbl order by aggCol"] [txn_mode=PESSIMISTIC] [err="runtime error: integer divide by zero\ngithub.com/pingcap/tidb/executor.(*recordSet).Next.func1\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/adapter.go:141\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:965\nruntime.panicdivide\n\t/usr/local/go/src/runtime/panic.go:191\ngithub.com/pingcap/tidb/util/selection.randomPivot\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:85\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:43\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.introselect\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:50\ngithub.com/pingcap/tidb/util/selection.Select\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/util/selection/selection.go:27\ngithub.com/pingcap/tidb/executor/aggfuncs.percentile\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggfuncs/func_percentile.go:43\ngithub.com/pingcap/tidb/executor/aggfuncs.(*percentileOriginal4Int).AppendFinalResult2Chunk\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggfuncs/func_percentile.go:172\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).appendResult2Chunk\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1376\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).consumeCurGroupRowsAndFetchChild\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1359\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).consumeOneGroup\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1296\ngithub.com/pingcap/tidb/executor.(*StreamAggExec).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/aggregate.go:1266\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/executor.go:285\ngithub.com/pingcap/tidb/executor.(*SortExec).fetchRowChunks\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/sort.go:228\ngithub.com/pingcap/tidb/executor.(*SortExec).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/sort.go:112\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/executor.go:285\ngithub.com/pingcap/tidb/executor.(*recordSet).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/executor/adapter.go:145\ngithub.com/pingcap/tidb/server.(*tidbResultSet).Next\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/driver_tidb.go:305\ngithub.com/pingcap/tidb/server.(*clientConn).writeChunks\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1993\ngithub.com/pingcap/tidb/server.(*clientConn).writeResultset\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1941\ngithub.com/pingcap/tidb/server.(*clientConn).handleStmt\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1835\ngithub.com/pingcap/tidb/server.(*clientConn).handleQuery\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1681\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:1215\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/conn.go:978\ngithub.com/pingcap/tidb/server.(*Server).onConn\n\t/home/jenkins/agent/workspace/build_tidb_multi_branch_master/go/src/github.com/pingcap/tidb/server/server.go:485\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"]
```

### 4. What is your TiDB version? (Required)

```
Release Version: v5.2.0-alpha-440-g17523d3da
Edition: Community
Git Commit Hash: 17523d3da8fe863979ae93277424f4ac5f0e9aa6
Git Branch: master
UTC Build Time: 2021-07-30 09:50:40
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Issue Number: close #26807 

Problem Summary:

`float64(28) / 100 * float64(100)` will produce `28.000000000000004` and ceil to 29 which over total len 28.

then random pivot index may cause issue's panic

### What is changed and how it works?

What's Changed, How it Works:

it's better to multiple by int and only div in float to avoid this.

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
Fix APPROX_PERCENTILE random panic for some percent value
```

### CODE REVIEW

**Review by cfzjywxk — APPROVED** [review]
/lgtm

File: executor/aggfuncs/func_percentile.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ var (
 
 func percentile(data sort.Interface, percent int) int {
 	// Ordinal rank k = Ceil(P / 100 * N)
-	k := int(math.Ceil(float64(data.Len()) / 100 * float64(percent)))
```
Comment (leiysky) [inline executor/aggfuncs/func_percentile.go:42]: Do the multiply first may cause overflow.

By the way, I've just filed a PR to fix this but find you've get it done 🤣 . https://github.com/pingcap/tidb/compare/master...leiysky:fix/approx_percentile?expand=1

My main idea is using decimal to calculate the precise `k` value here, FYI.

Comment (lysu) [reply executor/aggfuncs/func_percentile.go:42]: good catch~ how about `p/100` first, then len * [a num <= 1] should not great then len 😆

Comment (leiysky) [reply executor/aggfuncs/func_percentile.go:42]: Sounds good to me.

**Review by leiysky — APPROVED** [review]
LGTM

**Pull request merged**